packages/orm/src/collections/hasManyThrough.ts

```diff
@@ -0,0 +1,20 @@
+import { Collection, CustomCollection, currentlyInstantiatingEntity, Entity, getLens, Lens, loadLens } from "../index";
+
+/**
+ * Creates a CustomCollection that will walk across references in the object graph.
+ *
+ * I.e. An Author "has many reviews" through the `author -> books -> reviews` relation.
+ *
+ * Because this is based on `CustomCollection`, it will work in populates, i.e. `em.populate(author, "reviews")`.
+ */
+export function hasManyThrough<T extends Entity, U extends Entity>(
+  lens: (lens: Lens<T>) => Lens<U, U[]>,
+): Collection<T, U> {
+  const entity: T = currentlyInstantiatingEntity as T;
+  return new CustomCollection<T, U>(entity, {
+    load: async (entity) => {
+      await loadLens(entity, lens);
+    },
+    get: () => getLens(entity, lens) as U[],
```

Comment:
I kinda expected this return type of `getLens` to already be `U[]` given the `Lens<U, U[]>` type up above...

Reply:
ditto last comment